pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -853,7 +907,12 @@ cdef class _Timedelta(timedelta):
 
     def __hash__(_Timedelta self):
         if self._has_ns():
-            return hash(self.value)
+            # Note: this does *not* satisfy the invariance
+            #  td1 == td2 \\Rightarrow hash(td1) == hash(td2)
+            #  if td1 and td2 have different _resos. timedelta64 also has this
+            #  non-invariant behavior.
+            #  see GH#44504
+            return self.value
```

Comment:
Should this still be `hash(self.value)`? Maybe someone is relying on the hash being the same across different versions of pandas?

Reply:
yep, fixed.